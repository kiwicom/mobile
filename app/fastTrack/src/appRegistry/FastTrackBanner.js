// @flow strict

import * as React from 'react';
import {
  SimpleCard,
  Text,
  TextButton,
  StyleSheet,
} from '@kiwicom/mobile-shared';
import { Translation } from '@kiwicom/mobile-localization';
import { defaultTokens } from '@kiwicom/mobile-orbit';

import FastTrackModal from '../components/FastTrackModal';

type Props = {|
  +height: number,
|};

type State = {|
  +isModalVisible: boolean,
|};

class FastTrackBanner extends React.Component<Props, State> {
  state = {
    isModalVisible: false,
  };

  onOpenModal = () => {
    this.setState({
      isModalVisible: true,
    });
  };

  onCloseModal = () => {
    this.setState({
      isModalVisible: false,
    });
  };

  render() {
    return (
      <SimpleCard style={{ height: this.props.height }}>
        <Text style={style.title}>
          <Translation passThrough="Fast Track Banner test" />
        </Text>
        <TextButton
          type="primary"
          onPress={this.onOpenModal}
          title={<Translation passThrough="Open Modal" />}
        />
        <FastTrackModal
          isVisible={this.state.isModalVisible}
          onOpenModal={this.onOpenModal}
          onCloseModal={this.onCloseModal}
        />
      </SimpleCard>
    );
  }
}

export default FastTrackBanner;

const style = StyleSheet.create({
  title: {
    marginBottom: 16,
    textAlign: 'center',
    fontWeight: 'bold',
    color: defaultTokens.colorTextPrimary,
  },
});
