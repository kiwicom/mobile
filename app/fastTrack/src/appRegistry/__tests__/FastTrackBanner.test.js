// @flow

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import FastTrackBanner from '../FastTrackBanner';

describe('FastTrackBanner', () => {
  test('Should render correctly', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<FastTrackBanner bookingId={123} kwAuthToken="x.y.z" />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
