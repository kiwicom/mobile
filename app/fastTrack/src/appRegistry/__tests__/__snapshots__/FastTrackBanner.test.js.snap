// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FastTrackBanner Should render correctly 1`] = `
<PrivateApiRendererWithContext
  authHeaderKey="KW-Auth-Token"
  query={[Function]}
  render={[Function]}
  variables={
    Object {
      "bookingId": 123,
    }
  }
/>
`;
